### DISCUSSION TITLE
How to convert a matrix to edge attribute and edge index

### DISCUSSION THREAD

**Discussion by riskiem** [pyg-team/pytorch_geometric#2976] [Q&A]
I have a matrix as follows:

```
array([[0, 7, 4, 7],
       [7, 0, 9, 4],
       [4, 2, 0, 7],
       [7, 5, 8, 0]])
```

I want to convert this to the edge index and edge attribute for GNN. So for example for edge [1,2] I want the attribute to be 7. I am not sure how to go about this in pyotrch.

**Comment by wsad1** [comment]
One approach would be to use [torch.nonzero ](https://pytorch.org/docs/stable/generated/torch.nonzero.html)
```
adj = torch.tensor([[0, 7, 4, 7],
       [7, 0, 9, 4],
       [4, 2, 0, 7],
       [7, 5, 8, 0]])
edge_index = torch.nonzero(adj,)
edge_attr = adj[torch.nonzero(adj, as_tuple= True)]
```

**Reply by rusty1s** [reply]
There also exists [`torch_geometric.utils.dense_to_sparse`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dense_to_sparse).
